### ISSUE TITLE
Long shot: Hero Shop integration possible?

### ISSUE THREAD

**Issue by Vitae666** [MatthieuBonne/CQAutomater#48]
Just something that shows what heroes are in Hero shop and what they cost. With maybe a button next to each to be able to purchase? At least the UM one.
Have a surplus of UM's because I'm hoarding them right now and even tho I check as often as possible, I'm sure I'm JUST missing the UM heroes.
When I had 8K, the hero was 10K. When I go 10K the hero was 12K. And now I have 19K and no damned hero in sight for ages :(
If not, damn :(

**Comment by MatthieuBonne** [comment]
I should add that in http://dcouv.fr/cq.php ! Added to my todo.

**Comment by MatthieuBonne** [comment]
Your wishes are granted, my master. :)
http://dcouv.fr/cq.php?cal

**Comment by Vitae666** [comment]
Only Master here is you!
Will get it after stocking tp for 8hrs. LOL

**Comment by Vitae666** [comment]
Oh wait...I was talking about in CQA....
Display somewhere (Events tab?) showing that there's a hero avail for UM purchase and a button to make the purchase. With a display of how much it costs and how much you have on hand. Along with maybe the current level of the hero if you own it already so you don't spend 12K UM by mistake.
Hell, I'd prefer an autobuy if you have enough UM AND you don't already own the hero.

**Comment by MatthieuBonne** [comment]
I see. Well, now that LTO data is loaded in database, I just have to query it from CQA and code an autobuy (I'll need the json call for hero shop purchases, if you can get it for me).
I'll do it, but for now I must finish API calls to remove SQL stuff from CQA : all data will go through web requests, which means less MySQL connections from the outside of my server, no more SQL auth data in Secret.cs, and basically a cleaner code. ^^

**Comment by Vitae666** [comment]
I...I kinda need that in English...
How do I get it? What do I need to do? What needs to be open?

"I'll need the json call for hero shop purchases, if you can get it for me"

**Comment by MatthieuBonne** [comment]
It's in the network tab of a developer's console (in Chrome or Firefox). E.g. to open a chest I send this request : 
            var request = new ExecuteCloudScriptRequest()
            {
                RevisionSelection = CloudScriptRevisionOption.Live,
                FunctionName = "open",
                FunctionParameter = new { kid = kongID, mode = chestMode }
            };
Afaik hero shop would be like this but I have never tried : 
            var request = new ExecuteCloudScriptRequest()
            {
                RevisionSelection = CloudScriptRevisionOption.Live,
                FunctionName = "lto",
                FunctionParameter = new { offer = offer }
            };
Thing is, one has to buy something to test the request. Don't worry, I'll find a way. ^^

**Comment by MatthieuBonne** [comment]
Next commit (today) includes a first try of LTO purchases, only for first slot of AH tab, and only for heroes you don't own (so it's just a lvl0 -> lvl1 thing). Please test ! :)

**Comment by Vitae666** [comment]
Downloaded. Maybe I'm missing it? AH looks the same to me.

**Comment by LudoKressh** [comment]
I don't see anything either.

**Comment by MatthieuBonne** [comment]
For now you won't see anything, but if you ask to bid, say, up to 18k on Guy, and Guy appears in hero shop for 15k, CQA will (hopefully) buy from the shop.
I admit it's poor UX but I just wanted to test the purchase call. Now we should find a better display... Any idea ?

**Comment by LudoKressh** [comment]
maybe by writing it in the auction tab:

"If a hero is available in the store for less than the maximum bid, CQA will buy him from the store."

**Comment by Vitae666** [comment]
One problem there is that with having it in AH1, or any AH# is that you
might have had a chance at winning an auction.
I explained what I was thinking of poorly.
Ok, so you want to win Mother or any of those ridiculous ones that CC
players get from auction easy. The timer on them is 24 to 48 hrs. Up pops
Mother in the shop. You now have the option of hoping to win after waiting
48hrs or just spending likely more than what would win an auction.
And I thought there were heroes added to the shop that weren't in auction.
At. Least the newer ones for a month or so.
Was the reason my suggestion was for a button on its own.
But I know, as a test feature better to use what exists.
BTW, I've been checking the calendar and I swear, the heroes on there never
appear 😣

**Comment by Vitae666** [comment]
Well, willing to bet that I screwed up something. APontus was on the Hero Shop for 10K and wanted to get it.
Opened Network tab. Reloaded CQ to get a fresh page load. Set the tab to show only "Other" because it was the only one other than "All" that was showing json things.
Went to the Hero Shop. Bought APontus.

The only json file listed anywhere is beacon.min.js:23 that comes from https://beacon.s-onetag.com/beacon.min.js
[beacon.min.js.txt](https://github.com/MatthieuBonne/CQAutomater/files/4499910/beacon.min.js.txt)
But that looks useless.

I did a search for "offer" and there IS a bit of something with "offer" in it in the CosmosQuest.min.js file
[CosmosQuest.min.js.txt](https://github.com/MatthieuBonne/CQAutomater/files/4499919/CosmosQuest.min.js.txt)

Just really don't know what I was supposed to be doing or looking for :(
